### PR TITLE
Redeploy teleportation after audit on testnets & Deploy script

### LINKED ISSUE #1076: Redeploy Teleportation & add supported routes after 2nd audit complete
## Issue Type

We should redeploy teleportation contracts on testnet once newest audit feedback is being implemented. 
For now we would need to run services with old abi's. 

```
[ x] bug report
[ ] feature request
```

### PR DESCRIPTION
**Pull request by wsdt**
:clipboard: Resolves #1076, related to #1034 & #1083

## Overview

Redeploy Teleportation on all testnets, verified, update documentation & teleportation service. 
Since we changed the chainId datatype due to the first audit, the event signatures etc won't match for the gateway api etc. 

## Changes
- Update documentation
- Update addresses in teleportation service
- Redeploy Teleportation on all testnets
- Add deploy script to also conveniently have the routes somewhere configured (as it would be very cumbersome otherwise to keep track of it)
- Add EIP1155 support for Teleportation service to make it work on Goerli, etc.
- Upgrade ethereumjs-tx library & refactoring

## Testing
Actually redeployed everything (will also redeploy AWS services now)

### CODE REVIEW

File: packages/boba/contracts/.gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 src/contract-artifacts.ts
+teleportation_deploy_logs/*.txt
```
Comment (wsdt) [inline packages/boba/contracts/.gitignore:2]: If you want me to push the route logs, kindly let me know.

**Review by souradeep-das — APPROVED** [review]

**Comment by wsdt** [comment]
@Boyuan-Chen could you add these addresses to the AddressManager? Thank you

**Comment by boyuan-chen** [comment]
> @Boyuan-Chen could you add these addresses to the AddressManager? Thank you

Sure! Did we redeploy the proxy contract?

**Comment by wsdt** [comment]
> > @Boyuan-Chen could you add these addresses to the AddressManager? Thank you
> 
> Sure! Did we redeploy the proxy contract?

yes made a complete fresh deploy

**Comment by boyuan-chen** [comment]
> > > @Boyuan-Chen could you add these addresses to the AddressManager? Thank you
> > 
> > 
> > Sure! Did we redeploy the proxy contract?
> 
> yes made a complete fresh deploy

Done. I also registered them on networks.

**Comment by wsdt** [comment]
> > > > @Boyuan-Chen could you add these addresses to the AddressManager? Thank you
> > > 
> > > 
> > > Sure! Did we redeploy the proxy contract?
> > 
> > 
> > yes made a complete fresh deploy
> 
> Done. I also registered them on networks.

thx!

File: packages/boba/teleportation/package.json [hunk 2]
```diff
@@ -38,12 +38,12 @@
       "@aws-sdk/client-kms": "^3.363.0",
       "@eth-optimism/common-ts": "0.2.2",
       "@eth-optimism/contracts": "0.5.11",
+      "@ethereumjs/tx": "4.2.0",
       "@ethersproject/keccak256": "^5.7.0",
       "asn1.js": "^5.4.1",
       "bcfg": "^0.2.1",
       "bn.js": "^5.2.1",
       "dotenv": "^8.6.0",
-      "ethereumjs-tx": "^2.1.2",
```
Comment (wsdt) [inline packages/boba/teleportation/package.json:46]: package discontinued
-> @ethereumjs/tx should be used everywhere

**Comment by wsdt** [comment]
closing as #1083 is based on this branch (dependency)

**Pull request closed without merging**